apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/extract/RifFilesProcessor.java

```diff
@@ -64,363 +50,292 @@ public final class RifFilesProcessor {
-      final var rifParser = new SNFClaimParser();
-      isGrouped = true;
-      recordParser =
-          (fileEvent, csvRecords) -> buildSNFClaimEvent(fileEvent, csvRecords, rifParser);
-    } else if (file.getFileType() == RifFileType.HOSPICE) {
-      final var rifParser = new HospiceClaimParser();
-      isGrouped = true;
-      recordParser =
-          (fileEvent, csvRecords) -> buildHospiceClaimEvent(fileEvent, csvRecords, rifParser);
-    } else if (file.getFileType() == RifFileType.HHA) {
-      final var rifParser = new HHAClaimParser();
-      isGrouped = true;
-      recordParser =
-          (fileEvent, csvRecords) -> buildHHAClaimEvent(fileEvent, csvRecords, rifParser);
-    } else if (file.getFileType() == RifFileType.DME) {
-      final var rifParser = new DMEClaimParser();
-      isGrouped = true;
-      recordParser =
-          (fileEvent, csvRecords) -> buildDMEClaimEvent(fileEvent, csvRecords, rifParser);
-    } else {
-      throw new UnsupportedRifFileTypeException("Unsupported file type:" + file.getFileType());
-    }
-
-    /*
-     * Use the CSVParser to drive a Stream of grouped CSVRecords
-     * (specifically, group by claim ID/lines).
-     */
-    CsvRecordGrouper grouper =
-        new ColumnValueCsvRecordGrouper(isGrouped ? file.getFileType().getIdColumn() : null);
-    Iterator<List<CSVRecord>> csvIterator = new CsvRecordGroupingIterator(csvParser, grouper);
-    Spliterator<List<CSVRecord>> spliterator =
-        Spliterators.spliteratorUnknownSize(csvIterator, Spliterator.ORDERED | Spliterator.NONNULL);
-    Stream<List<CSVRecord>> csvRecordStream =
-        StreamSupport.stream(spliterator, false)
-            .onClose(
-                () -> {
-                  try {
-                    /*
-                     * This will also close the Reader and InputStream that the
-                     * CSVParser was consuming.
-                     */
-                    csvParser.close();
-                  } catch (IOException e) {
-                    LOGGER.warn("Unable to close CSVParser", e);
-                  }
-                });
-
-    /* Map each record group to a single RifRecordEvent. */
-    Stream<RifRecordEvent<?>> rifRecordStream =
-        csvRecordStream.map(
-            csvRecordGroup -> {
-              Timer.Context parsingTimer =
-                  rifFileEvent
-                      .getEventMetrics()
-                      .timer(MetricRegistry.name(getClass().getSimpleName(), "recordParsing"))
-                      .time();
-              RifRecordEvent<?> recordEvent =
-                  parseRifRecord(rifFileEvent, csvRecordGroup, recordParser);
-              // OK if an exception prevents the close because a failed parse is not a valid sample.
-              parsingTimer.close();
-
-              return recordEvent;
-            });
-
-    return new RifFileRecords(rifFileEvent, rifRecordStream);
+    RifFile rifFile = rifFileEvent.getFile();
+    RifFileParser rifFileParser =
```

Comment:
Talk about a reduction in code....and much cleaner!

Reply:
Thanks! I t was a perfect fit for a switch expression.